### PR TITLE
Adding physics engine integration docs

### PR DESCRIPTION
**Pull request by Nibba2018**
Solves #312 .

TODO:

* [x] Code Blocks, tables and Images need formatting.

### CODE REVIEW

**Comment by skoudoro** [comment]
Nice! Exciting, thanks @Nibba2018.

Since many physics libraries could be connected, we might need to have a section for physics, and `pybullet` as a subsection. But you do not need to worry about that. we will reshape it later.

**Comment by Nibba2018** [comment]
Hi @skoudoro , Where should I save the images and gifs for this doc?

**Comment by skoudoro** [comment]
Hi @Nibba2018, you can save the gif in [this repo](https://github.com/fury-gl/fury-communication-assets).

Concerning the images, you can save them in `fury/docs/source/_static/images`. 

Is it soon ready to go? Should I start to review it? Thanks for the feedback

**Comment by Nibba2018** [comment]
Hello @skoudoro , great to have you back. Hope you had an amazing holiday season!

It's almost ready, apart from the linking of headers. I have made a PR regarding the gifs. Once that is merged I will do some testing and then I will ask for a review. :+1:

**Comment by Nibba2018** [comment]
@skoudoro The PR is now ready for review.

File: docs/source/fury-pybullet.rst
Comment (skoudoro) [inline docs/source/fury-pybullet.rst:49]: This block looks like a local table of content (toc). I suppose it can be replaced by 
```
.. contents:: Table of Contents (or other title)
    :local:
    :depth: 2
``` 

More information [here](https://docutils.sourceforge.io/docs/ref/rst/directives.html#table-of-contents). Can you give a try? it will make the maintenance easier

File: docs/source/fury-pybullet.rst
Comment (skoudoro) [inline docs/source/fury-pybullet.rst:1]: `.. _fury_pybullet:` is missing at the top of the file (before this title).

Also, FURY in capital everywhere

File: docs/source/fury-pybullet.rst
Comment (skoudoro) [inline docs/source/fury-pybullet.rst:57]:
```
.. note::
  **All elements are in SI units.**
```
instead.
more info [here](https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#colored-boxes-note-seealso-todo-and-warnings)

File: docs/source/fury-pybullet.rst
Comment (skoudoro) [inline docs/source/fury-pybullet.rst:59]: Not sure you need all these links if you have a `toc`

File: docs/source/fury-pybullet.rst
Comment (skoudoro) [inline docs/source/fury-pybullet.rst:64]: same comments as above

File: docs/source/fury-pybullet.rst
Comment (skoudoro) [inline docs/source/fury-pybullet.rst:121]: :py:class:\`.Scene()\`    instead of Scene

File: docs/source/fury-pybullet.rst
Comment (skoudoro) [inline docs/source/fury-pybullet.rst:131]: to remove, same as above

File: docs/source/fury-pybullet.rst
Comment (skoudoro) [inline docs/source/fury-pybullet.rst:168]: `.. note:: *Centers for the actor must be set to (0, 0, 0) or else the simulation will be offset by that particular value.*` instead

File: docs/source/fury-pybullet.rst
Comment (skoudoro) [inline docs/source/fury-pybullet.rst:170]: to remove because of toc

File: docs/source/fury-pybullet.rst
Comment (skoudoro) [inline docs/source/fury-pybullet.rst:181]: `..note:: *The second parameter is linkIndex which is for bodies having multiple links or joints. Passing -1 means applying changes to the base object.*`

File: docs/source/fury-pybullet.rst
Comment (skoudoro) [inline docs/source/fury-pybullet.rst:347]: update note syntax

File: docs/source/fury-pybullet.rst
Comment (skoudoro) [inline docs/source/fury-pybullet.rst:434]: update note syntax

**Review by skoudoro — COMMENTED** [review]
Hi @Nibba2018,

I had a nice holiday, Thank you. I hope it was good for you too and I wish you the best for this year!

Concerning this PR, see my comments below. Overall it looks good. However, the syntax can be simplified and improved. You will have to update all you `note` and use a local table of content (toc) instead of doing it manually.

We also need to see where to include this file.

Let me know when I can do a second review and thanks for doing that.

File: docs/source/fury-pybullet.rst
Comment (Nibba2018) [reply docs/source/fury-pybullet.rst:49]: Definitely, I was not aware of the toc directive.

**Comment by Nibba2018** [comment]
Thank you @skoudoro , I enjoyed my holidays too.

I have made the requested changes do have a look.

**Review by skoudoro — APPROVED** [review]
ok, looks better now and easier to maintain, thank you for the update!

Can you add your file in [index.rst](https://github.com/fury-gl/fury/blob/master/docs/source/index.rst)  in `Programming with FURY` section.

Then we can merge this PR. Thanks!

**Pull request merged**
